broker-cli/broker-daemon-client/config.js

```diff
@@ -1,25 +1,44 @@
 const path = require('path')
 const os = require('os')
 
+/**
+ * @type {String}
+ * @constant
+ * @default
+ */
+const MISSING_FILE_MESSAGE = 'Cannot find module'
+
 /**
  * Default configuration is in the root directory for the SparkSwap CLI
  * @type {Object}
  * @constant
  */
 const DEFAULT_CONFIG = require('../.sparkswap.default.js')
 
-try {
-  /**
-   * Attempt to load user-defined configuration in their home directory
-   * @type {Object}
-   */
-  var USER_CONFIG = require(path.resolve(os.homedir(), '.sparkswap.js'))
-} catch (e) {
-  USER_CONFIG = {}
-}
-
 /**
- * Export a configuration object with user configuration taking precedence over defaults
- * @type {Object}
+ * A wrapper around path resolution for a user defined `.sparkswap.js` configuration
+ * file
+ *
+ * We attempt to load the file and set default configuration values. We will also
+ * warn the user if we have failed to parse their user config file.
+ *
+ * @return {Object} config
  */
-module.exports = Object.assign({}, DEFAULT_CONFIG, USER_CONFIG)
+function loadConfig () {
+  try {
+    var config = require(path.resolve(os.homedir(), '.sparkswap.js'))
+  } catch (e) {
+    // We will only trigger an error if the `~/.sparkswap.js` configuration file
+    // exists on the user's machine. Otherwise we just silently set to a default
```

Comment:
We will warn the user only if the `~/.sparkswap.js` configuration file exists but could not be read. We will set the config to the default regardless of the error.